common/src/main/scala/net/psforever/objects/vehicles/VehicleControl.scala

```diff
@@ -73,11 +73,31 @@ class VehicleControl(vehicle : Vehicle) extends Actor
     .orElse(takesDamage)
     .orElse(canBeRepairedByNanoDispenser)
     .orElse {
-      case msg : Mountable.TryMount =>
+      case Vehicle.Ownership(None) =>
+        LoseOwnership()
+
+      case Vehicle.Ownership(Some(player)) =>
+        GainOwnership(player)
+
+      case msg @ Mountable.TryMount(player, seat_num) =>
         tryMountBehavior.apply(msg)
-        if(MountableObject.Seats.values.exists(_.isOccupied)) {
-          decaying = false
-          decayTimer.cancel
+        val obj = MountableObject
+        if(obj.Seats.values.exists(_.isOccupied)) {
+          if(seat_num == 0 && !obj.OwnerName.contains(player.Name)) {
+            //whatever vehicle was previously owned
+            vehicle.Zone.GUID(player.VehicleOwned) match {
+              case Some(v : Vehicle) =>
+                v.Actor ! Vehicle.Ownership(None)
+              case _ =>
+                player.VehicleOwned = None
+            }
+            LoseOwnership() //lose our current ownership
+            GainOwnership(player) //gain new ownership
+          }
+          else {
```

Comment:
Should this else block not be one level higher, so it's the else block for `obj.Seats.values.exists(_.isOccupied)`?

Reply:
If `obj.Seats.values.exists(_.isOccupied)` fails, the player that tried to mount the vehicle had failed to mount the vehicle.  We shouldn't stop the decay timer in this case.  Additionally, `seat_num == 0` just goes through a little extra driver-specific code that ultimately does the exact same thing as the `else` clause.  I will do a bit of switch up with this, however, so comments will be added for clarification.

Reply:
Ah yeah, that's a good point. I was thinking about the case where someone is seated in something other than the driver seat, do we still want to start the decay timer in that case?

Reply:
No, any player occupancy in the vehicle should suspend the decay timer.  That's why `TryDismount` checks `!decaying && obj.Owner.isEmpty && obj.Seats.values.forall(!_.isOccupied)` (are we already decaying? are we unowned? is no one seated anywhere?) before trying to restart it.

Reply:
Ah that's all good then. I missed the bit in `TryDismount`.